src/common.c

```diff
@@ -391,16 +394,48 @@ ly_mmap(struct ly_ctx *ctx, int fd, size_t *length, void **addr)
         LOGERR(ctx, LY_ESYS, "mmap() failed (%s).", strerror(errno));
         return LY_ESYS;
     }
-
+#else
+    /* On Windows, the mman-win32 mmap() emulation uses CreateFileMapping and MapViewOfFile, and these functions
+     * do not allow mapping more than "length of file" bytes for PROT_READ. Remapping existing mappings is not allowed, either.
+     * At that point the path of least reisstance is just reading the file in as-is. */
+    m = sb.st_size + 1;
+    char *buf = calloc(m, 1);
+    if (!buf) {
+        LOGERR(ctx, LY_ESYS, "ly_mmap: malloc() failed (%s).", strerror(errno));
+    }
+    *addr = buf;
+    *length = m;
+
+    lseek(fd, 0, SEEK_SET);
+    ssize_t to_read = m - 1;
+    while (to_read > 0) {
+        ssize_t n = read(fd, buf, to_read);
+        if (n == 0) {
+            return LY_SUCCESS;
+        } else if (n < 0) {
+            if (errno == EINTR) {
+                continue; // can I get this on Windows?
+            }
+            LOGERR(ctx, LY_ESYS, "ly_mmap: read() failed (%s).", strerror(errno));
+        }
+        to_read -= n;
+        buf += n;
+    }
+#endif
```

Comment:
I don't particularly like such a long code in `#ifdef` directly in the main source file. Maybe you could create a Windows-specific source file with functions called from the main source.

Reply:
Done.